starter-code/src/viking.js

```diff
@@ -1,11 +1,104 @@
 // Soldier
-function Soldier() {}
+// soldier = new Soldier(health, strength);
 
-// Viking
+
+class Soldier(){
+    consutrctuor(healthArg, strengthArg){
+        this.health = healthArg;
+        this.strength = strengthArg;
+    }
+    attack(){
+        return this.strength;
+    }
+    receiveDamage(amoungOfDamage){
+        this.health -= amountOfDamage;
+    }
+}
+
+// Viking 
 function Viking() {}
 
+
+class Viking extends Soldier{
+    constructor(theName, theHealth, theStrength){
+        super(theHealth, theStrength)
+        this.name = theName;
+    }
+
+    receiveDamage(amount){
+        super.receiveDamage(amount);
+        if(this.health > 0){
+            return `${this.name} has reeceived ${amount} points of damage`
+        } else {
+            return `${this.name} has died in act of combat`
+        }
+
+    }
+
+    battleCry(){
+        return "Odin owns you all!";
+    }
+
+
+}
+
 // Saxon
 function Saxon() {}
 
+class Saxon extends Soldier{
+    constructor(health, stregnth){
+        super(health, strength); 
+        if(this.health > 0){
```

Comment:
wrap all this if statement into a receiveDamage method and add the operation to subtract the health.

```class Saxon extends Soldier{
  constructor(health, strength){
    super(health, strength); 
  }
  receiveDamage(theDamage) {
    this.health -= theDamage
    if(this.health > 0){
      return `a Saxon has received ${amount} points of damage`
    } else {
      return `A Saxon has died in act of combat`
    }
  }
}
```

Reply:
also add the attack method